src/graph_1.py

```diff
@@ -23,6 +23,10 @@ def add_node(self, val):
 
     def add_edge(self, val1, val2, wt=0):
         """Add new edge to graph connecting nodes "val1" and "val2"."""
+        if val1 not in self.node_list:
+            self.add_node(val1)
+        if val2 not in self.node_list:
+            self.add_node(val2)
         self.edges[(val1, val2)] = wt
```

Comment:
This "weight" parameter shouldn't even be here. Every edge in this iteration of the graph is unweighted.